src/helpers/getErc725yContract/getErc725yContract.ts

```diff
@@ -0,0 +1,45 @@
+import { BytesLike, Provider, Signer, Wallet, isAddress, isAddressable } from 'ethers';
+import { ERC725Y, ERC725Y__factory } from '../..';
+import { INTERFACE_IDS } from '@lukso/lsp-smart-contracts';
+
+export async function getErc725yContract(erc725y: ERC725Y): Promise<ERC725Y>;
+export async function getErc725yContract(
+    erc725y: ERC725Y,
+    signer: Signer | Wallet | Provider,
+): Promise<ERC725Y>;
+export async function getErc725yContract(
+    erc725y: BytesLike | string,
+    signer: Signer | Wallet | Provider,
+): Promise<ERC725Y>;
+export async function getErc725yContract(
+    erc725y: ERC725Y | BytesLike | string,
+    signer?: Signer | Wallet | Provider,
+): Promise<ERC725Y> {
+    let erc725yContract: ERC725Y;
+
+    if (isAddress(erc725y)) {
+        erc725yContract = ERC725Y__factory.connect(erc725y, signer);
+    } else if (isAddressable(erc725y)) {
+        if (signer) {
+            erc725yContract = erc725y.connect(signer);
+        } else {
+            erc725yContract = erc725y;
+        }
+    } else {
+        throw new Error(
+            `The parameter \`erc725y\` is not a valid address nor a valid contract instance of \`ERC725Y\`. Value: '${erc725y}'`,
+        );
+    }
```

Comment:
Prefer return early pattern: https://medium.com/swlh/return-early-pattern-3d18a41bba8

 ```suggestion    
   if (!isAddress(erc725y) && !isAddressable(erc725y)) {
       throw new Error(
            `The parameter \`erc725y\` is not a valid address nor a valid contract instance of \`ERC725Y\`. Value: '${erc725y}'`,
        );
    } 

    let erc725yContract: ERC725Y;

    if (isAddress(erc725y)) {
        erc725yContract = ERC725Y__factory.connect(erc725y, signer);
    } else if (signer) {
         erc725yContract = erc725y.connect(signer);
    } else {
          erc725yContract = erc725y;
      }
    }
```

(I wrote it on GitHub textarea so might have wrong tab or pb but thats the idea